processing/src/main/java/org/apache/druid/query/QueryContexts.java

```diff
@@ -88,6 +88,8 @@ public class QueryContexts
   public static final String UNCOVERED_INTERVALS_LIMIT_KEY = "uncoveredIntervalsLimit";
   public static final String MIN_TOP_N_THRESHOLD = "minTopNThreshold";
   public static final String CATALOG_VALIDATION_ENABLED = "catalogValidationEnabled";
+  public static final String EXTENDED_FILTERED_SUM_REWRITE_ENABLED = "extendedFilteredSumRewrite";
```

Comment:
I see the new context parameter in this PR is undocumented and defaults to `true` (the old behavior, more performant, yet incorrect in the edge cases). IMO it's OK to have it be undocumented, but we should have a javadoc here explaining what is going on. Can you please add one and link it both to the rule (using `@link`) and also says something like: this is here for testing purposes, it is a known issue that the production configuration is incorrect in certain edge cases, and this context parameter can be removed once we have a way to do a rewrite of these case statements that is both high-performance and correct.